documentation/modules/ROOT/pages/ai/embeddings.adoc

```diff
@@ -0,0 +1,191 @@
+|
+|Name of the new record field which will container text embedding.
+If no value is specified, record value will contain only embedding itself.
+|===
+
+== Model provider configuration
+
+=== Hugging Face
+
+Embeddings provided by models available via link:https://huggingface.co//[Hugging Face].
+
+.Huggins Face embeddings configuration options
+[cols="30%a,25%a,45%a",subs="+attributes",options="header"]
+|===
+|Option
+|Default
+|Description
+
+|[[embeddings-huggingface-access-token]]xref:embeddings-huggingface-access-token[`embeddings.huggingface.access.token`]
+|
+|Hugging Face access token.
+|[[embeddings-huggingface-model-name]]xref:embeddings-huggingface-model-name[`embeddings.huggingface.model.name`]
+|
+|Name of the embedding model.
+List of available models can be obtained e.g. via REST API.
+You need to specify the provider in the REST call, e.g. link:https://huggingface.co/api/models?inference_provider=hf-inference[Hugging Face inference provider].
+|[[embeddings-huggingface-baseurl]]xref:embeddings-huggingface-baseurl[`embeddings.huggingface.baseUrl`]
+|https://api-inference.huggingface.co/
+|Base Hugging Face inference API URL.
+|[[embeddings-huggingface-operation-timeout-ms]]xref:embeddings-huggingface-operation-timeout-ms[`embeddings.huggingface.operation.timeout.ms`]
+|15,000 (15 seconds)
+|Maximum amount of time in milliseconds to wait for the embeddings reply.
+|===
+
+[NOTE]
+====
+Hugging Face link:https://huggingface.co/blog/inference-providers[started to provide embedding inference] also via link:https://huggingface.co/docs/inference-providers/en/index[external providers].
+However, external providers don't seem to be supported by langchain4j framework yet.
+Until it's supported by langchain4j, the only available inference provider is Hugging Face provider.
+====
+
+
+=== Ollama
+
+Supports any model provided by link:https://ollama.com/[Ollama] server.
+
+.Ollama embeddings configuration options
+[cols="30%a,25%a,45%a",subs="+attributes",options="header"]
+|===
+|Option
+|Default
+|Description
+
+|[[embeddings-ollama-url]]xref:embeddings-ollama-url[`embeddings.ollama.url`]
+|
+|URL of the Ollama server, including port number, e.g. `http://localhost:11434`.
+|[[embeddings-ollama-model-name]]xref:embeddings-ollama-model-name[`embeddings.ollama.model.name`]
+|
+|Name of the embedding model.
+|[[embeddings-ollama-operation-timeout-ms]]xref:embeddings-ollama-operation-timeout-ms[`embeddings.ollama.operation.timeout.ms`]
+|15,000 (15 seconds)
+|Maximum amount of time in milliseconds to wait for the embeddings reply.
+|===
+
+=== ONNX MiniLM
+
+Provides link:https://docs.langchain4j.dev/integrations/embedding-models/in-process[ONNX in-process] `all-minilm-l6-v2` model, which is included directly in the jar file.
+No other configuration besides the general one is needed.
+
+As no other infrastructure or remote requests are needed in this case, this model is suitable especially for prototyping and testing.
```

Comment:
```suggestion
This model is especially suited to prototyping and testing, because it does not depend on any external infrastructure or remote requests. 
```